AcadeMeData/test_register.py

```diff
@@ -0,0 +1,46 @@
+import pytest
+from AcadeMeData.forms import UserRegistrationForm
+from AcadeMeData.models import User
+
+
+@pytest.fixture
+def user_example():
+    user_data = {'username': "daniel", 'email': "user@example.com", 'password1': "D4587450", 'password2': "D4587450",
+                 'university': "Hebrew University", 'degree': "Computer Science"}
+    return user_data
+
+
+@pytest.fixture
+def user_bad_example():
+    # username can't be none
+    users_data = [
+        {'username': "", 'email': "user@example.com", 'password1': "good password", 'password2': "good password",
+         'university': "RU", 'degree': "CS"},
+        # invalid email
+        {'username': "good username", 'email': "user", 'password1': "good password", 'password2': "good password",
+         'university': "RU", 'degree': "CS"},
+        # passqords don't match
+        {'username': "good username", 'email': "user@example.com", 'password1': "good password",
+         'password2': "bad password", 'university': "RU", 'degree': "CS"}]
+    return users_data
+
+
+@pytest.mark.django_db
+class TestSignUp:
+    def test_sign_up_invalid(self, user_bad_example):
+        for user_data in user_bad_example:
+            invalid = False
+            form = UserRegistrationForm(data=user_data)
+            try:
+                form.save()
+            except ValueError:
+                invalid = True
+            assert invalid
+
+    def test_sign_up_valid(self, user_example):
```

Comment:
@Omeramsc @Yarboa 
this is the test that failed - for some reason even though all values are legal for registration - form.is_valid() - returns false. could use some help with this.

Reply:
Did you try creating the form and validate in the shell?

Reply:
@Yarboa now it is fixed.

Reply:
Yes you had to try shell
>>> form.is_valid()
False
>>> form.errors
{'university': ['Select a valid choice. Hebrew University is not one of the available choices.'], 'degree': ['Select a valid choice. Computer Science is not one of the available choices.']}
>>>